contracts/Delay.sol

```diff
@@ -4,44 +4,59 @@ import "@aragon/os/contracts/apps/AragonApp.sol";
 import "@aragon/os/contracts/common/IForwarder.sol";
 import "@aragon/os/contracts/lib/math/SafeMath64.sol";
 
-
-contract Delay is AragonApp, IForwarder {
+import "@aragon/os/contracts/apps/disputable/DisputableAragonApp.sol";
+
+/*
+TODO: Remove the below once UI is updated.
+Possible states:
+Active
+Paused (/challenged)
+Executable
+Closed (also Executable and Agreement action closed)
+Executed (also Closed, determined by searching for ID in ExecutedScript event)
+Cancelled (also Closed, determined by searching for ID in ExecutionCancelled event)
+*/
+
+contract Delay is DisputableAragonApp, IForwarder {
     using SafeMath64 for uint64;
 
-    bytes32 public constant SET_DELAY_ROLE = keccak256("SET_DELAY_ROLE");
-    bytes32 public constant DELAY_EXECUTION_ROLE = keccak256("DELAY_EXECUTION_ROLE");
-    bytes32 public constant PAUSE_EXECUTION_ROLE = keccak256("PAUSE_EXECUTION_ROLE");
-    bytes32 public constant RESUME_EXECUTION_ROLE = keccak256("RESUME_EXECUTION_ROLE");
-    bytes32 public constant CANCEL_EXECUTION_ROLE = keccak256("CANCEL_EXECUTION_ROLE");
+    /**
+        bytes32 public constant SET_DELAY_ROLE = keccak256("SET_DELAY_ROLE");
+        bytes32 public constant DELAY_EXECUTION_ROLE = keccak256("DELAY_EXECUTION_ROLE");
+    */
+    bytes32 public constant SET_DELAY_ROLE = 0x2b56821903fce674a357ec19f2dd583428c0cc74e6be302c348b3a3ffa5b4f66;
+    bytes32 public constant DELAY_EXECUTION_ROLE = 0x68a7ca9b0904e026378fb888f3be95ada2a0c0b11f58c40530c1a383c0f99ce9;
 
     string private constant ERROR_NO_SCRIPT = "DELAY_NO_SCRIPT";
     string private constant ERROR_CAN_NOT_EXECUTE = "DELAY_CAN_NOT_EXECUTE";
-    string private constant ERROR_CAN_NOT_PAUSE = "DELAY_CAN_NOT_PAUSE";
-    string private constant ERROR_SCRIPT_EXECUTION_PASSED = "DELAY_SCRIPT_EXECUTION_PASSED";
-    string private constant ERROR_CAN_NOT_RESUME = "DELAY_CAN_NOT_RESUME";
     string private constant ERROR_CAN_NOT_FORWARD = "DELAY_CAN_NOT_FORWARD";
 
+    enum DelayedScriptStatus {
+        Active,              // A delayed script that has been reported to Agreements
+        Paused,              // A delayed script that is being challenged by Agreements
+        Cancelled,           // A delayed script that has been cancelled by Agreements
+        Executed             // A delayed script that has been executed
+    }
+
     struct DelayedScript {
-        uint64 executionTime;
+        uint64 executionFromTime;
         uint64 pausedAt;
         bytes evmCallScript;
+        uint256 actionId;
+        DelayedScriptStatus delayedScriptStatus;
```

Comment:
You may want to have the enum next to the `uint64` variables

Reply:
Pending discussion regarding backwards compatibility I'll do this for now.